### PR TITLE
Optimize lib/Mount to PHP8

### PR DESCRIPTION
**Pull request by solracsf**
PHP8 optimizations.

### CODE REVIEW

File: lib/Mount/CacheRootPermissionsMask.php [hunk 1]
```diff
@@ -36,11 +36,15 @@ public function __construct(ICache $cache, int $mask) {
 
 	protected function formatCacheEntry($entry) {
 		$path = $entry['path'];
-		$isRoot = $path === '' || (strpos($path, '__groupfolders') === 0 && count(explode('/', $path)) === 2);
+		$isRoot = $path === '' || (str_starts_with($path, '__groupfolders/') && count(explode('/', $path)) === 2);
+
+		$entry['scan_permissions'] ??= $entry['permissions'];
```
Comment (come-nc) [inline lib/Mount/CacheRootPermissionsMask.php:41]: You will get a warning if `$entry['permissions']` is not defined, and this is changing the logic of the function which was only doing that if isRoot is true before?

File: lib/Mount/GroupFolderStorage.php [hunk 2]
```diff
@@ -52,34 +52,23 @@ public function getFolderId(): int {
...
-		}
-		return $storage->scanner;
+		$storage ??= $this;
+
+		$scanner = match (true) {
+			$storage->instanceOfStorage(ObjectStoreStorage::class) => $storage->scanner ??= new ObjectStoreScanner($storage),
+			default => $storage->scanner ??= new Scanner($storage),
+		};
+
+		return $scanner;
```
Comment (come-nc) [inline lib/Mount/GroupFolderStorage.php:72]: This is less readable than before I find.

File: lib/Mount/MountProvider.php [hunk 3]
```diff
@@ -126,29 +126,23 @@ public function getFoldersForUser(IUser $user): array {
...
-		}, $foldersWithAcl);
+		$foldersWithAcl = array_filter($folders, fn(array $folder) => $folder['acl']);
+		$aclRootPaths = array_map(fn(array $folder) => $this->getJailPath($folder['folder_id']), $foldersWithAcl);
+
 		$aclManager = $this->aclManagerFactory->getACLManager($user, $this->getRootStorageId());
 		$aclManager->preloadPaths($aclRootPaths);
 
-		return array_values(array_filter(array_map(function ($folder) use ($user, $loader, $conflicts, $aclManager) {
-			// check for existing files in the user home and rename them if needed
+		array_walk($folders, function (&$folder) use ($user, $loader, $conflicts, $aclManager) {
```
Comment (come-nc) [inline lib/Mount/MountProvider.php:138]: Why replace the array_map with an array_walk?

File: lib/Mount/MountProvider.php [hunk 4]
```diff
@@ -188,7 +184,7 @@ private function getCurrentUID(): ?string {
 		}
 
 		$user = $this->userSession->getUser();
-		return $user ? $user->getUID() : null;
+		return $user?->getUID() ?? null;
```
Comment (come-nc) [inline lib/Mount/MountProvider.php:187]:
```suggestion
		return $user?->getUID();
```

**Review by come-nc — COMMENTED** [review]
All in all, apart from `fn` usage, I’m not really convinced by the changes, a lot of them are sacrifying readability for compactness, especially all the ternary instead of if/else.

**Comment by solracsf** [comment]
After reading it myself i also believe this is less readable...
Closing.

**Pull request closed without merging**